gallery/experiments/example_halfset_pipeline_10028.py

```diff
@@ -0,0 +1,150 @@
+# Example configuration.
+#
+# Use of GPU is expected for a large configuration.  If running on a
+# less capable machine, or simply experimenting, it is strongly
+# recommended to reduce the problem size by altering ``img_size``,
+# ``n_classes``, ``n_nbor``, ``max_rows`` etc.
+
+img_size = 179  # Downsample the images/reconstruction to a desired resolution
+n_classes = 3000  # How many class averages to compute.
+n_nbor = 50  # How many neighbors to stack
+starfile_in = "10028/data/shiny_2sets_fixed9.star"
+data_folder = "."  # This depends on the specific starfile entries.
+pixel_size = 1.34
+fsc_cutoff = 0.143
+
+# %%
+# Load and split Source data
+# --------------------------
+#
+# ``RelionSource`` is used to access the experimental data via a `STAR` file.
+
+# Create a source object loading all the experimental images
+src = RelionSource(
+    starfile_in, pixel_size=pixel_size, max_rows=None, data_folder=data_folder
+)
+
+# Split the data into two sets.
+# This example uses evens and odds for simplicity, but random indices
+# can also be used with similar results.
+
+srcA = src[::2]
+srcB = src[1::2]
+
+# A dictionary can systematically organize our inputs and outputs for both sets.
+pipelines = {
+    "A": {
+        "input": srcA,
+    },
+    "B": {
+        "input": srcB,
+    },
+}
+
+# %%
+# Dual Pipelines
+# --------------
+# Preprocess by downsampling, correcting for CTF, and applying noise
+# correction.  After preprocessing, class averages are generated then
+# automatically selected for use in reconstruction.
+#
+# Each of the above steps are performed totally independently
+# for each dataset, first for A, then for B.
+#
+# Caching is used throughout for speeding up large datasets on high memory machines.
+
+for src_id, pipeline in pipelines.items():
+
+    src = pipeline["input"]
+
+    # Downsample the images
+    logger.info(f"Set the resolution to {img_size} X {img_size}")
+    src = src.downsample(img_size).cache()
+
+    # Use phase_flip to attempt correcting for CTF.
+    logger.info(f"Perform phase flip to {len(src)} input images for set {src_id}.")
+    src = src.phase_flip().cache()
+
+    # Normalize the background of the images.
+    src = src.normalize_background().cache()
+
+    # Estimate the noise and whiten based on the estimated noise.
+    src = src.legacy_whiten().cache()
+
+    # Optionally invert image contrast.
+    logger.info("Invert the global density contrast")
+    src = src.invert_contrast().cache()
+
+    # Now perform classification and averaging for each class.
+    logger.info("Begin Class Averaging")
+
+    avgs = LegacyClassAvgSource(src, n_nbor=n_nbor)
+    avgs = avgs[:n_classes].cache()
+
+    # Common Line Estimation
+    logger.info("Begin Orientation Estimation")
+    oriented_src = OrientedSource(avgs)
+
+    # Volume Reconstruction
+    logger.info("Begin Volume reconstruction")
+
+    # Setup an estimator to perform the back projection.
+    estimator = MeanEstimator(oriented_src)
```

Comment:
This is more of a point of discussion to bring up later. It is sometimes beneficial to reconstruct from the original (raw) images as opposed to the class averages in order to remove some of the bias (lowpass filtering) induced by the class averaging process. This would basically take the orientations assigned to the class averages and assign those to the original images, then use those to reconstruct. I'm not sure this is what was done in the JSB paper, but it's something to think about.

Reply:
Understood.  This was not done.  I do have some soft evidence the author attempted to use FIRM on the oriented class averages as is done here.

Reply:
Ok got it.